### PR TITLE
[Merged by Bors] - feat: `tfae` tactics

### PR DESCRIPTION
**Pull request by thorimur**
We implement a basic version of #2061 to attain mathlib parity:
* `tfae_have` followed by e.g. a tactic block, in parallel to mathlib `have` syntax (note: `tfae_have 1 → 2 := ...` is not supported yet, as it was not supported by the original `tfae_have` tactic, and would constitute new syntax)
```
example : TFAE [P, Q, R] := by
  tfae_have h : 1 → 2
  { /- proof of P → Q -/ }
  tfae_have 2 → 3
  { /- proof of Q → R -/ }
  ...
```
* `tfae_finish`, which looks through the local context and simply tries to prove each implication in a cycle via `solve_by_elim`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/TFAE.lean [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  tfae_have : 1 → 2
+  { /- proof of P → Q -/ }
+  tfae_have : 2 → 1
+  { /- proof of Q → P -/ }
+  tfae_have : 2 ↔ 3
+  { /- proof of Q ↔ R -/ }
+  tfae_finish
+```
+-/
+syntax (name := tfaeHave) "tfae_have " (ident " : ")? num impArrow num : tactic
```
Comment (jcommelin) [inline Mathlib/Tactic/TFAE.lean:60]: I'm cargo-culting, but should this be:
```suggestion
syntax (name := tfaeHave) "tfae_have " (binderIdent " : ")? num impArrow num : tactic
```

Comment (thorimur) [reply Mathlib/Tactic/TFAE.lean:60]: (Also, sorry for the delayed response; I fell prey to GitHub's "pending" label and thought I had taken all the action I needed to. 🙃)

image

Comment (thorimur) [reply Mathlib/Tactic/TFAE.lean:60]: `impArrow` question resolved [on zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/refactoring.20mathport.20syntax.3F).

**Comment by jcommelin** [comment]
@thorimur As far as I can tell, this is looking pretty good and stable. If you agree, let's just merge this. If there are small bugs, we can fix them later.

bors d+

**Pull request closed without merging**